integration/network/service_test.go

```diff
@@ -106,9 +101,6 @@ func TestServiceWithIngressNetwork(t *testing.T) {
 	err = client.ServiceRemove(context.Background(), serviceID)
 	require.NoError(t, err)
 
-	poll.WaitOn(t, serviceIsRemoved(client, serviceID), pollSettings)
-	poll.WaitOn(t, noTasks(client), pollSettings)
```

Comment:
If we don't check if the service (plus tasks?) are all removed here, this test may not be testing what it was created for (i.e.: check that the ingress network is not removed when removing services)

Reply:
Idea was to check service is removed instead task as we are testing service create/remove. Thats why I added new API. Let me update TestServiceWithIngressNetwork. Thanks for pointing  it out.

Reply:
BTW actual test that was written by Chris Telfer was to verify Ingress network absence after service remove. So test will fail if it is not met.  Daniel mentioned service clean up will happen automatically when we tear down the swarm . so we dont need to check this explicitly.  Anyways I update the with new API that TestServiceWithIngressNetwork with my newly added noServices check to make sure we cover that .